common/src/main/java/com/wynntils/features/combat/AutoAttackFeature.java

```diff
@@ -0,0 +1,122 @@
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ArmSwingEvent;
+import com.wynntils.mc.event.ArmSwingEvent.ArmSwingContext;
+import com.wynntils.mc.event.ChangeCarriedItemEvent;
+import com.wynntils.mc.event.SetSlotEvent;
+import com.wynntils.mc.event.TickEvent;
+import com.wynntils.mc.event.UseItemEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.models.spells.event.SpellEvent;
+import com.wynntils.models.spells.type.SpellDirection;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.wynn.ItemUtils;
+import net.minecraft.client.player.LocalPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.SubscribeEvent;
+
+@ConfigCategory(Category.COMBAT)
+public class AutoAttackFeature extends Feature {
+    private static final int TICKS_PER_ATTACK = 2;
+    private static final int SPELL_TIMEOUT_TICKS = 60;
+
+    private enum WeaponStatus {
+        UNKNOWN,
+        NOT_USABLE,
+        USABLE
+    }
+
+    private WeaponStatus weaponStatus = WeaponStatus.UNKNOWN;
+    private int lastSelectedSlot;
+    private int preventWrongCast = Integer.MIN_VALUE;
+
+    @SubscribeEvent
+    public void onChangeCarriedItemEvent(ChangeCarriedItemEvent event) {
+        weaponStatus = WeaponStatus.UNKNOWN;
+    }
+
+    @SubscribeEvent
+    public void onSetSlotEvent(SetSlotEvent.Post event) {
+        if (event.getSlot() == McUtils.inventory().selected) {
+            weaponStatus = WeaponStatus.UNKNOWN;
+        }
+    }
+
+    @SubscribeEvent
+    public void onSwing(ArmSwingEvent event) {
+        if (event.getActionContext() != ArmSwingContext.ATTACK_OR_START_BREAKING_BLOCK) return;
+        if (event.getHand() != InteractionHand.MAIN_HAND) return;
+        if (Models.Character.getClassType() != ClassType.ARCHER) return;
+        lastSelectedSlot = McUtils.inventory().selected;
+        preventWrongCast = McUtils.player().tickCount + SPELL_TIMEOUT_TICKS;
+    }
+
+    @SubscribeEvent
+    public void onUseItem(UseItemEvent event) {
+        if (Models.Character.getClassType() == ClassType.ARCHER) return;
+        lastSelectedSlot = McUtils.inventory().selected;
+        preventWrongCast = McUtils.player().tickCount + SPELL_TIMEOUT_TICKS;
+    }
+
+    @SubscribeEvent
+    public void onSpellCastCompleted(SpellEvent.Completed event) {
+        preventWrongCast = Integer.MIN_VALUE;
+    }
+
+    private boolean isHoldingUsableWeapon() {
+        ItemStack heldItem = McUtils.player().getItemInHand(InteractionHand.MAIN_HAND);
+        if (!ItemUtils.isWeapon(heldItem)) return false;
+        if (ItemUtils.getItemName(heldItem).contains("Unidentified")) return false;
+        for (StyledText lore : LoreUtils.getLore(heldItem)) {
+            if (lore.contains("✖")) return false;
+        }
+        return true;
+    }
```

Comment:
Still not a fan of this entire function - it's a duplicate of information that already exists.
Since we now save the usability of the weapon, can you look into using parsing the weapon as a WynnItem and using .meetsActualRequirements()?

Example at InventoryModel.java Line 77-81, if you need help.